internal/binutils/disasm.go

```diff
@@ -62,7 +69,7 @@ func matchSymbol(names []string, start, end uint64, r *regexp.Regexp, address ui
 		return names
 	}
 	for _, name := range names {
-		if r.MatchString(name) {
+		if r == nil || r.MatchString(name) {
```

Comment:
Do we pass r as nil from anywhere? Ah, you do know in `f.Symbols(nil, 0)` in the test. I think you can just do `f.Symbols(regexp.MustCompile(tc.sym), 0)`?.

Reply:
No but [the comments for the `Symbols()` interface](https://github.com/google/pprof/blob/43bdac25ec8dcf95b98bba82e7a162ff406a20b3/internal/plugin/plugin.go#L149) say it can be nil.